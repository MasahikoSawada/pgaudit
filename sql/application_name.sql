-- Create test table
CREATE TABLE appname1 (col int);
CREATE TABLE appname2 (col int);

-- Check if the READ class audit log is emitted only during
-- application_name is 'appname1'.
SELECT * FROM appname1;
SET application_name TO 'appname1';
SELECT * FROM appname1; -- must be logged
SET application_name TO DEFAULT;
SELECT * FROM appname1;

-- Check if the ERROR class audit log is emitted only during
-- application_name is 'appname2'.
SELECT err_col FROM appname2; -- error
SET application_name TO 'appname2';
SELECT err_col FROM appname2; -- error, must be logged
SET application_name TO DEFAULT ;
SELECT err_col FROM appname2; -- error
