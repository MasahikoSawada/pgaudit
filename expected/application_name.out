-- Create test table
CREATE TABLE appname1 (col int);
CREATE TABLE appname2 (col int);
-- Check if the READ class audit log is emitted only during
-- application_name is 'appname1'.
SELECT * FROM appname1;
 col 
-----
(0 rows)

SET application_name TO 'appname1';
SELECT * FROM appname1; -- must be logged
NOTICE:  AUDIT: SESSION,1,1,READ,,regression,[local],appname1,,,,,SELECT,TABLE,public.appname1,SELECT * FROM appname1;,<not logged>
 col 
-----
(0 rows)

SET application_name TO DEFAULT;
SELECT * FROM appname1;
 col 
-----
(0 rows)

-- Check if the ERROR class audit log is emitted only during
-- application_name is 'appname2'.
SELECT err_col FROM appname2; -- error
ERROR:  column "err_col" does not exist
LINE 1: SELECT err_col FROM appname2;
               ^
SET application_name TO 'appname2';
SELECT err_col FROM appname2; -- error, must be logged
NOTICE:  AUDIT: SESSION,,,ERROR,,regression,[local],appname2,,,42703,column "err_col" does not exist,,,,,
ERROR:  column "err_col" does not exist
LINE 1: SELECT err_col FROM appname2;
               ^
SET application_name TO DEFAULT ;
SELECT err_col FROM appname2; -- error
ERROR:  column "err_col" does not exist
LINE 1: SELECT err_col FROM appname2;
               ^
